pontoon/base/models.py

```diff
@@ -561,16 +561,33 @@ def get_chart_data(stats):
     )
 
 
-def get_locales_with_stats(project):
+def get_locales_with_stats():
+    """Add chart data to locales."""
+
+    locales = Locale.objects.filter(stats__isnull=False).distinct()
+
+    for locale in locales:
+        stats = Stats.objects.filter(
+            resource__entity__obsolete=False,
+            locale=locale).distinct()
+
+        locale.chart = get_chart_data(stats)
```

Comment:
`get_chart_data` looks like a method that should be on the `Locale` class, that takes an optional `project` argument for filtering by project. Then you could add a manager method to the Locale class that prefetches the necessary stats that the function uses to reduce the number of queries, like what you did with the entity translations.